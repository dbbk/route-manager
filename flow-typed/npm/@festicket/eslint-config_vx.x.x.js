// flow-typed signature: 564086c0274f153b4c37ac5c954f219e
// flow-typed version: <<STUB>>/@festicket/eslint-config_v^1.1.5/flow_v0.63.1

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@festicket/eslint-config'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@festicket/eslint-config' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */


// Filename aliases
declare module '@festicket/eslint-config/index' {
  declare module.exports: $Exports<'@festicket/eslint-config'>;
}
declare module '@festicket/eslint-config/index.js' {
  declare module.exports: $Exports<'@festicket/eslint-config'>;
}
