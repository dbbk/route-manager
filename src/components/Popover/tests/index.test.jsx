// @flow

import React from 'react';
import { mount } from 'enzyme';
import { Popover } from '@festicket/react-ui-components';

describe('<Popover />', () => {
  test('with initial props', () => {
    const tree = mount(
      <Popover render={() => <div>Content</div>}>
        <button>Toggle</button>
      </Popover>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('with showInitially props', () => {
    const tree = mount(
      <Popover showInitially render={() => <div>Content</div>}>
        <button>Toggle</button>
      </Popover>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('it toggles a popover on click', () => {
    const TriggerButton = props => <button {...props}>Toggle</button>;

    const tree = mount(
      <Popover render={() => <div>Content</div>}>
        <TriggerButton />
      </Popover>,
    );

    const trigger = tree.find('TriggerButton');
    const popoverElement = tree.find('PopoverTransition').getDOMNode();

    expect(popoverElement.style.display).toBe('none');

    trigger.simulate('click');
    expect(popoverElement.style.display).toBe('inline-block');

    trigger.simulate('click');
    expect(popoverElement.style.display).toBe('none');
  });

  test('it updates ARIA attributes on popover show and hide', () => {
    const TriggerButton = props => <button {...props}>Toggle</button>;

    const tree = mount(
      <Popover render={() => <div>Content</div>}>
        <TriggerButton />
      </Popover>,
    );

    expect(tree.find('TriggerButton').prop('aria-haspopup')).toBe('true');
    expect(tree.find('TriggerButton').prop('aria-expanded')).toBe('false');

    tree.find('TriggerButton').simulate('click');

    expect(tree.find('TriggerButton').prop('aria-expanded')).toBe('true');

    tree.find('TriggerButton').simulate('click');
    expect(tree.find('TriggerButton').prop('aria-expanded')).toBe('false');
  });
});
