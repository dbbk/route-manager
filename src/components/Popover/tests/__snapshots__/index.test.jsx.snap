// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> with initial props 1`] = `
.c1 {
  display: block;
  -webkit-text-align: left;
  text-align: left;
  padding: 20px;
  box-shadow: 0 2px 40px 0;
}

@media (min-width:0px) and (max-width:568px) {
  .c0 {
    left: 0 !important;
    right: 0 !important;
  }
}

<Popover
  render={[Function]}
>
  <div>
    <button
      aria-expanded="false"
      aria-haspopup="true"
      onClick={[Function]}
    >
      Toggle
    </button>
    <PopoverTransition
      shown={false}
      style={
        Object {
          "left": 0,
          "top": 0,
        }
      }
    >
      <StyledCSSTransition
        active={false}
        activeStyle={
          Object {
            "opacity": 1,
            "transform": "translateY(0)",
          }
        }
        defaultStyle={
          Object {
            "display": "none",
            "opacity": 0,
            "transform": "translateY(15px)",
          }
        }
        enterInitStyle={
          Object {
            "display": "inline-block",
            "opacity": 0,
            "transform": "translateY(15px)",
          }
        }
        enterStyle={
          Object {
            "opacity": Array [
              1,
            ],
            "transform": Array [
              "translateY(0)",
            ],
          }
        }
        leaveStyle={
          Object {
            "opacity": Array [
              0,
            ],
            "transform": Array [
              "translateY(15px)",
            ],
          }
        }
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "top": 0,
          }
        }
      >
        <CSSTransition
          active={false}
          activeStyle={
            Object {
              "opacity": 1,
              "transform": "translateY(0)",
            }
          }
          className="c0"
          defaultStyle={
            Object {
              "display": "none",
              "opacity": 0,
              "transform": "translateY(15px)",
            }
          }
          enterInitStyle={
            Object {
              "display": "inline-block",
              "opacity": 0,
              "transform": "translateY(15px)",
            }
          }
          enterStyle={
            Object {
              "opacity": Array [
                1,
              ],
              "transform": Array [
                "translateY(0)",
              ],
            }
          }
          leaveStyle={
            Object {
              "opacity": Array [
                0,
              ],
              "transform": Array [
                "translateY(15px)",
              ],
            }
          }
          style={
            Object {
              "left": 0,
              "position": "absolute",
              "top": 0,
            }
          }
        >
          <div
            className="c0"
            onTransitionEnd={[Function]}
            style={
              Object {
                "display": "none",
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transform": "translateY(15px)",
              }
            }
          >
            <span
              key="workaround"
              style={
                Object {
                  "opacity": 0.9,
                }
              }
            />
            <PopoverContent
              innerRef={[Function]}
              key=".0"
            >
              <div
                className="c1"
              >
                <div>
                  Content
                </div>
              </div>
            </PopoverContent>
          </div>
        </CSSTransition>
      </StyledCSSTransition>
    </PopoverTransition>
  </div>
</Popover>
`;

exports[`<Popover /> with showInitially props 1`] = `
.c1 {
  display: block;
  -webkit-text-align: left;
  text-align: left;
  padding: 20px;
  box-shadow: 0 2px 40px 0;
}

@media (min-width:0px) and (max-width:568px) {
  .c0 {
    left: 0 !important;
    right: 0 !important;
  }
}

<Popover
  render={[Function]}
  showInitially={true}
>
  <div>
    <button
      aria-expanded="true"
      aria-haspopup="true"
      onClick={[Function]}
    >
      Toggle
    </button>
    <PopoverTransition
      shown={true}
      style={
        Object {
          "left": 0,
          "top": 0,
        }
      }
    >
      <StyledCSSTransition
        active={true}
        activeStyle={
          Object {
            "opacity": 1,
            "transform": "translateY(0)",
          }
        }
        defaultStyle={
          Object {
            "display": "none",
            "opacity": 0,
            "transform": "translateY(15px)",
          }
        }
        enterInitStyle={
          Object {
            "display": "inline-block",
            "opacity": 0,
            "transform": "translateY(15px)",
          }
        }
        enterStyle={
          Object {
            "opacity": Array [
              1,
            ],
            "transform": Array [
              "translateY(0)",
            ],
          }
        }
        leaveStyle={
          Object {
            "opacity": Array [
              0,
            ],
            "transform": Array [
              "translateY(15px)",
            ],
          }
        }
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "top": 0,
          }
        }
      >
        <CSSTransition
          active={true}
          activeStyle={
            Object {
              "opacity": 1,
              "transform": "translateY(0)",
            }
          }
          className="c0"
          defaultStyle={
            Object {
              "display": "none",
              "opacity": 0,
              "transform": "translateY(15px)",
            }
          }
          enterInitStyle={
            Object {
              "display": "inline-block",
              "opacity": 0,
              "transform": "translateY(15px)",
            }
          }
          enterStyle={
            Object {
              "opacity": Array [
                1,
              ],
              "transform": Array [
                "translateY(0)",
              ],
            }
          }
          leaveStyle={
            Object {
              "opacity": Array [
                0,
              ],
              "transform": Array [
                "translateY(15px)",
              ],
            }
          }
          style={
            Object {
              "left": 0,
              "position": "absolute",
              "top": 0,
            }
          }
        >
          <div
            className="c0"
            onTransitionEnd={[Function]}
            style={
              Object {
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "top": 0,
                "transform": "translateY(0)",
              }
            }
          >
            <span
              key="workaround"
              style={
                Object {
                  "opacity": 0.9,
                }
              }
            />
            <PopoverContent
              innerRef={[Function]}
              key=".0"
            >
              <div
                className="c1"
              >
                <div>
                  Content
                </div>
              </div>
            </PopoverContent>
          </div>
        </CSSTransition>
      </StyledCSSTransition>
    </PopoverTransition>
  </div>
</Popover>
`;
